students/arun_nalla/session05/Lesson5_Mailroom_Part3.py

```diff
@@ -0,0 +1,99 @@
+#!usr/bin/env python
+
+donors_list = [('Aron King', [1, 2]), ('Becky Patty', [3000, 1000]),
+               ('Charlie Lee', [666, 222]), ('Thomas Dave', [202, 90, 450]), ('Nancy Crow', [100, 200, 300])]
+# input from Mairoom_Part1. Convert into a dict
+
+donors_dict = {x: y for x, y in donors_list}
+
+def thanks_email():
+    while True:
+        name_input = input("Enter full name of a donor>>>").title()
+        if name_input == 'List': print(list(donors_dict))
+        else:
+            if name_input in donors_dict:
+                d = {}
+                d['name'] = name_input
+                while True:
+                    try:
+                        donation_amount = int(input('Enter donation amount: '))
+                    except ValueError:
+                        print ('Please enter current value, entry should be an integer.')
+                    else:
+                        donors_dict[name_input].append(donation_amount)
+                        d['donation'] = donation_amount
+                        print("\nDear {name},\n\tThank you for generous donation of ${donation:.2f} to ZYX charity\n"
+                          "Sincerely,\nZYX Team".format(**d))
+                        return
+            else:
+                while True:
+                    d = {}
+                    d['name'] = name_input
+                    try:
+                        donation_amount = int(input('Enter donation amount: '))
+                    except ValueError:
+                        print ('Please enter a current value, entry should be an integer.')
+                    else:
+                        donors_dict[name_input] = [donation_amount]
+                        d['donation'] = donation_amount
+                        #dict_format(seq,name_input)
+                        print("\nDear {name},\n\tThank you for generous donation of ${donation:.2f} to ZYX charity\n"
+                          "Sincerely,\nZYX Team".format(**d))
+                        return
```

Comment:
Almost identical code (lines 17-27 and lines 29-42), which is not desirable at all. I think I might have already mentioned this as well... You could have used `donors_dict.setdefault(name_input, []).append(donation_amount)` to unify the only different part (line 23 and line 37).